frontend/src/store/ToastState.ts

```diff
@@ -0,0 +1,72 @@
+import { ref } from 'vue';
+import axios from 'axios';
+import { useToast } from 'primevue/usetoast';
+
+// the toastError state is just used to store the error before enter the router
+export const toastError = ref('');
+export const setToastError = (error: string) => {
+    toastError.value = error;
+};
+export const clearToastError = () => {
+    toastError.value = '';
+};
+
+// the primevue toast can only be injected in setup or functional component,
+// in order to reuse the toast service,
+// we have to define it in the way as below
+// https://stackoverflow.com/questions/72425297/vue-warn-inject-can-only-be-used-inside-setup-or-functional-components
```

Comment:
Thanks @MCatherine1994  for finding this out and adding this comment, very useful!!
From the reference, it is probably the same issue I encountered just few weeks ago when using `Provide/Inject` that they only work for Vue components; and same issue happens for the `PrimeVue Toast`.
I see the reference mentions in the future, they want to make Toast able to handle all cases, that time we can simplify more on using toast service.